test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java

```diff
@@ -65,19 +64,19 @@
     private final ClientYamlSuiteRestSpec restSpec;
     private final Map<NodeSelector, RestClient> restClients = new HashMap<>();
     private final Version esVersion;
-    private final CheckedConsumer<RestClientBuilder, IOException> clientBuilderConsumer;
+    private final CheckedSupplier<RestClientBuilder, IOException> clientBuilderBuilder;
 
     ClientYamlTestClient(
             final ClientYamlSuiteRestSpec restSpec,
             final RestClient restClient,
             final List<HttpHost> hosts,
             final Version esVersion,
-            final CheckedConsumer<RestClientBuilder, IOException> clientBuilderConsumer) {
+            final CheckedSupplier<RestClientBuilder, IOException> clientBuilderBuilder) {
```

Comment:
can we rename the instance member and corresponding arguments to something like clientBuilderSupplier or clientBuilderWithSniffedNodes?

Reply:
Sounds good to me!